src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -2549,20 +2566,19 @@ Connection connectInternal(Properties propsIn,
                         throw new SQLServerException(
                                 SQLServerException.getErrString("R_NoUserPasswordForActiveServicePrincipal"), null);
                     }
+                }
 
-                    if ((!activeConnectionProperties.getProperty(SQLServerDriverStringProperty.USER.toString())
-                            .isEmpty()
-                            || !activeConnectionProperties
-                                    .getProperty(SQLServerDriverStringProperty.PASSWORD.toString()).isEmpty())
-                            && (!activeConnectionProperties
-                                    .getProperty(SQLServerDriverStringProperty.AAD_SECURE_PRINCIPAL_ID.toString())
-                                    .isEmpty()
-                                    || !activeConnectionProperties.getProperty(
-                                            SQLServerDriverStringProperty.AAD_SECURE_PRINCIPAL_SECRET.toString())
-                                            .isEmpty())) {
-                        throw new SQLServerException(SQLServerException.getErrString("R_BothUserPasswordandDeprecated"),
-                                null);
-                    }
+                if ((!activeConnectionProperties.getProperty(SQLServerDriverStringProperty.USER.toString()).isEmpty()
+                        || !activeConnectionProperties.getProperty(SQLServerDriverStringProperty.PASSWORD.toString())
+                                .isEmpty())
+                        && (!activeConnectionProperties
+                                .getProperty(SQLServerDriverStringProperty.AAD_SECURE_PRINCIPAL_ID.toString()).isEmpty()
+                                || !activeConnectionProperties
+                                        .getProperty(
+                                                SQLServerDriverStringProperty.AAD_SECURE_PRINCIPAL_SECRET.toString())
+                                        .isEmpty())) {
+                    throw new SQLServerException(SQLServerException.getErrString("R_BothUserPasswordandDeprecated"),
+                            null);
```

Comment:
Just wondering about this block here. Since it was moved out the of the check `if (authenticationString.equalsIgnoreCase(SqlAuthentication.ACTIVE_DIRECTORY_SERVICE_PRINCIPAL...`, will something like msi auth where the user/password and clientId/secret isn't specified cause it to erroneously throw `R_BothUserPasswordandDeprecated`?